src/avk.cpp

```diff
@@ -4934,6 +4934,17 @@ namespace avk
 		rewire_config_and_create_graphics_pipeline(result);
 		return result;
 	}
+
+	renderpass&& root::replace_render_pass_for_pipeline(graphics_pipeline& aPipeline, renderpass aNewRenderPass)
+	{
+		if (aPipeline->mRenderPass.is_shared_ownership_enabled()) {
+			aNewRenderPass.enable_shared_ownership();
+		}
+
+		std::swap(aPipeline->mRenderPass, aNewRenderPass);
+
+		return std::move(aNewRenderPass); // now old render pass

```

Comment:
This is returning a reference to a local variable. (I wonder if the compiler did issue a warning?)

Best practice would be to just rely on named return value optimization (NRVO) and return by value!

The question arises if it would maybe be better to not return anything and instead, just take a reference to `renderpass& aNewRenderPass` and swap directly the passed `renderpass&`. However, that would have to be documented well and might lead to more confusion for users of the method. So if you think that two additional moves are worth the increased clarity, you may leave the function's general structure like that.

Reply:
changed to return by value. oops.